projects/storefrontlib/src/cms-components/checkout/guards/payment-details-set.guard.ts

```diff
@@ -29,23 +30,31 @@ export class PaymentDetailsSetGuard implements CanActivate {
       );
     }
 
-    if (checkoutStep && checkoutStep.disabled) {
-      return of(true);
-    }
-
-    return this.checkoutDetailsService
-      .getPaymentDetails()
-      .pipe(
-        map((paymentDetails) =>
-          paymentDetails && Object.keys(paymentDetails).length !== 0
+    return combineLatest([
+      this.paymentTypeService.getSelectedPaymentType(),
+      this.checkoutDetailsService.getPaymentDetails(),
+    ]).pipe(
+      filter(([selected, _]) => selected !== undefined),
+      tap(([selected, _]) =>
+        this.checkoutStepService.disableEnableStep(
+          CheckoutStepType.PAYMENT_DETAILS,
+          selected === this.paymentTypeService.ACCOUNT_PAYMENT
+        )
+      ),
+      map(([_, paymentDetails]) => {
```

Comment:
Same thing about destructuring arrays, I believe you could use `[, paymentDetails]` instead of `[_, paymentDetails]` if you don't need the array's first element.  Your call.